projects/plugins/jetpack/extensions/blocks/premium-content/buttons/edit.js

```diff
@@ -59,14 +55,22 @@ function ButtonsEdit( { context, subscribeButton, setSubscribeButtonPlan } ) {
 	// Keep in sync the plan selected on the Premium Content block with the plan selected on the Recurring Payments
 	// inner block acting as a subscribe button.
 	useEffect( () => {
-		if ( ! planId || ! subscribeButton ) {
+		if ( ! planIds || ! subscribeButton ) {
```

Comment:
I think this will always evaluate to true. You might want `planIds.length === 0`